app/src/main/java/org/fossasia/openevent/activities/MainActivity.java

```diff
@@ -86,6 +99,23 @@ protected void onCreate(Bundle savedInstanceState) {
 
     }
 
+    public void WiFiDialog(final Activity activity){
+        downloadProgress.setVisibility(View.GONE);
+        AlertDialog.Builder builder = new AlertDialog.Builder(activity);
+        builder.setTitle(getString(R.string.net_unavailable))
+                .setMessage(getString(R.string.turn_on))
+                .setPositiveButton(R.string.action_settings, new DialogInterface.OnClickListener() {
+                    @Override
+                    public void onClick(DialogInterface dialog, int which) {
+                        Intent setNetworkIntent = new Intent(Settings.ACTION_SETTINGS);
+                        startActivity(setNetworkIntent);
+                        activity.finish();
```

Comment:
I don't think you need to call activity.finish(), I know you are doing this to force download the information the next time the person comes back to the app, but I think that should be handled from a connectivity change receiver because network can change anytime and you need that.